frontend/packages/console-shared/src/components/dropdown/dropdown-with-switch/DropdownWithSwitchToggle.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react';
+import { MenuToggle, Popper } from '@patternfly/react-core';
+
+const DropdownWithSwitchToggle: React.FC<DropdownWithSwitchToggleProps> = ({
+  isDisabled,
+  isFullWidth,
+  isOpen,
+  label,
+  menu,
+  menuRef,
+  onToggle,
+}) => {
+  const toggleRef = React.useRef(null);
+  const containerRef = React.useRef(null);
+  const handleMenuKeys = (event) => {
+    if (menuRef.current) {
+      if (event.key === 'Escape') {
+        onToggle(false);
+        toggleRef.current.focus();
+      }
+      if (!menuRef.current?.contains(event.target) && event.key === 'Tab') {
+        onToggle(false);
+      }
+    }
+  };
+  const handleMenuClick = (event) => {
+    if (
+      menuRef.current &&
+      !menuRef.current?.contains(event.target) &&
+      !toggleRef.current.contains(event.target)
+    ) {
+      onToggle(false);
+    }
+  };
+  React.useEffect(() => {
+    window.addEventListener('keyup', handleMenuKeys);
+    window.addEventListener('click', handleMenuClick);
+    return () => {
+      window.removeEventListener('keyup', handleMenuKeys);
+      window.removeEventListener('click', handleMenuClick);
+    };
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []); // This needs to be run only on component mount/unmount
```

Comment:
You cannot do this from an idiomatic point of view -- JS Closures are going to hurt your code stability over the duration/use of this component. You need to properly support a dependency array.

1. Move the menu key/click handlers inside useEffect
2. Take your `onToggle` inline function in `DropdownWithSwitch` to a `useCallback`
3. Pass the `useCallback` value into this component as `onToggle`
4. Make `menuRef` & `onToggle` part of this dependency array

That should solve the reference issue... and obv remove the lint disable and comment.

Reply:
The why behind this can be found here: https://github.com/andrewballantyne/code-examples/blob/main/example3/README.md